pyface/tests/test_font.py

```diff
@@ -0,0 +1,285 @@
+        self.assertEqual(font.weight, 'normal')
+        self.assertEqual(font.stretch, 100)
+        self.assertEqual(font.style, 'normal')
+        self.assertEqual(font.variants, set())
+
+    def test_typical(self):
+        font = Font(
+            family=['Helvetica', 'sans-serif'],
+            size='large',
+            weight='demibold',
+            stretch='condensed',
+            style='italic',
+            variants={'small-caps', 'underline'},
+        )
+
+        self.assertEqual(font.family, ['Helvetica', 'sans-serif'])
+        self.assertEqual(font.size, 14.0)
+        self.assertEqual(font.weight, 'demibold')
+        self.assertEqual(font.weight_, 600)
+        self.assertEqual(font.stretch, 75)
+        self.assertEqual(font.style, 'italic')
+        self.assertEqual(font.variants, {'small-caps', 'underline'})
+
+    def test_family_sequence(self):
+        font = Font(family=('Helvetica', 'sans-serif'))
+        self.assertEqual(font.family, ['Helvetica', 'sans-serif'])
+
+    def test_variants_frozenset(self):
+        font = Font(variants=frozenset({'small-caps', 'underline'}))
+        self.assertEqual(font.variants, {'small-caps', 'underline'})
+
+    def test_str(self):
+        font = Font()
+
+        description = str(font)
+
+        self.assertEqual(description, "12pt default")
+
+    def test_str_typical(self):
+        font = Font(
+            family=['Comic Sans', 'decorative'],
+            size='large',
+            weight='demibold',
+            stretch='condensed',
+            style='italic',
+            variants={'small-caps', 'underline'},
+        )
+
+        description = str(font)
+
+        self.assertEqual(
+            description,
+            "italic small-caps underline demibold 75% 14pt "
+            "'Comic Sans', decorative"
+        )
+
+    def test_repr(self):
+        font = Font()
+
+        text = repr(font)
+
+        # this is little more than a smoke check, but good enough
+        self.assertTrue(text.startswith('Font('))
+
+    def test_repr_typical(self):
+        font = Font(
+            family=['Helvetica', 'sans-serif'],
+            size='large',
+            weight='demibold',
+            stretch='condensed',
+            style='italic',
+            variants={'small-caps', 'underline'},
+        )
+
+        text = repr(font)
+
+        # this is little more than a smoke check, but good enough
+        self.assertTrue(text.startswith('Font('))
+
+    def test_to_toolkit(self):
+        font = Font()
+
+        # smoke test
+        toolkit_font = font.to_toolkit()
+
+    def test_to_toolkit_typical(self):
+        font = Font(
+            family=['Helvetica', 'sans-serif'],
+            size='large',
+            weight='demibold',
+            stretch='condensed',
+            style='italic',
+            variants={'small-caps', 'underline'},
+        )
+
+        # smoke test
+        toolkit_font = font.to_toolkit()
+
+    def test_from_toolkit(self):
```

Comment:
can we also add roundtrip tests here? the roundtrip tests for Color reminded me of their lack here.

Reply:
No, unfortunately there isn't 100% fidelity in the round-trips: Qt can lose 2nd choice font families, and Wx loses some font weights, so you aren't guaranteed to get the same thing back.

Reply:
Although, now that I think about it, it should be round-trip stable toolkit -> pyface -> toolkit.

Reply:
It turns out if you do pyface -> toolkit -> pyface then you get something that can should be able to do another -> toolkit -> pyface without changing.  I've added tests for this.

Reply:
Or, alternatively, what is true on my system isn't true on random CI system with who know s what fonts or lack thereof.  I'm going to remove those tests, as they will be essentially impossible to guarantee that they are right on every system.

Reply:
> ... , as they will be essentially impossible to guarantee that they are right on every system.

hmm. fair enough.